[Fix] cleanup and fixes

MaxWasUnavailable opened the pull request:
General cleanup using Rider's code cleanup functionality.

Additionally:
- Updated UI assets
- Fixed namespace mismatches from the config -> json achievement refactor
- Added docstrings to public methods

MaxWasUnavailable committed:
Fix version

MaxWasUnavailable committed:
Move HUDController to static instance

MaxWasUnavailable committed:
Use range indexer

MaxWasUnavailable committed:
Make StringHelper public

MaxWasUnavailable committed:
Move method from Truncate to StringHelper

MaxWasUnavailable committed:
Make PascalToSnakeCase an extension method & add docs

MaxWasUnavailable committed:
Move to patches module

MaxWasUnavailable committed:
Target only netstandard2.1

Support for more modern language features

MaxWasUnavailable committed:
Refactor to add docstrings & cleanup logging

MaxWasUnavailable committed:
Cleanup ModEntry

MaxWasUnavailable committed:
Add additional logging line

MaxWasUnavailable committed:
Change to conditional properties

MaxWasUnavailable committed:
Add conditional UI initialisation based on asset loading

MaxWasUnavailable committed:
Use conditional fields & add file exists check

MaxWasUnavailable committed:
Make private properties into private fields

MaxWasUnavailable committed:
Make private properties into private fields

MaxWasUnavailable committed:
Cleanup nullability & set active to false in init

MaxWasUnavailable committed:
Use method groups

MaxWasUnavailable committed:
Change to conditionals & proper private field names

MaxWasUnavailable committed:
Add ... if truncated

MaxWasUnavailable committed:
Set explicit not null

MaxWasUnavailable committed:
Apply Rider cleanup

MaxWasUnavailable committed:
Remove excessive debug logging

MaxWasUnavailable committed:
Fix spelling error

MaxWasUnavailable committed:
Apply Rider cleanup

MaxWasUnavailable committed:
Cleanup conditions

MaxWasUnavailable committed:
Cleanup predicates

MaxWasUnavailable committed:
Cleanup Serialization

Corrected namespace, so a lot of extra changes

MaxWasUnavailable committed:
Apply rider cleanup to Triggers

MaxWasUnavailable committed:
Cleanup JSON

Fixed namespaces & applied rider cleanup

MaxWasUnavailable committed:
Move UI patches to UI folder

MaxWasUnavailable committed:
Resolve compiler warnings

MaxWasUnavailable committed:
Remove unnecessary verbose debug line

MaxWasUnavailable committed:
Move patching to separate method

MaxWasUnavailable committed:
Expand credits

MaxWasUnavailable committed:
Patch explicitly rather than all

MaxWasUnavailable committed:
Add logging

MaxWasUnavailable committed:
Update assets

MaxWasUnavailable committed:
Try-catch UI loading

MaxWasUnavailable committed:
Close achievements on quick menu close

MaxWasUnavailable commented:
Would like your review, @Kesomannen!

MaxWasUnavailable committed:
Remove verbose debug logs

MaxWasUnavailable committed:
Fix version

Kesomannen commented on LethalAchievements/Events/Patches/EnemyDamageSource.cs:
```diff
@@ -11,17 +10,14 @@ namespace LethalAchievements.Events.Patches;
     private static int _currentEnemyHitFrame = int.MinValue;
 
-    private static readonly MethodInfo _enemyDamagePathMethod = AccessTools.Method(typeof(EnemyDamageSource), nameof(EnemyDamagePatch));
+    private static readonly MethodInfo EnemyDamagePathMethod =
```
Why should this be capitalized?

MaxWasUnavailable replied:
MSDN convention

Kesomannen replied:
According to [this](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions) it should technically be prefixed with s_

Kesomannen replied:
Also just noticed theres a typo 😄. It's supposed to be patch, not path.

MaxWasUnavailable replied:
> According to [this](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions) it should technically be prefixed with s_

Hmm, it seems like [static readonly are a matter of debate](https://stackoverflow.com/questions/4577358/private-static-readonly-field-capitalization). [StyleCop implies it should be capitalised](https://github.com/StyleCop/StyleCop/blob/master/Project/Docs/Rules/SA1311.html), however. Since the rest of the project adheres to StyleCop (implicitly, through Rider's default configurations), I feel it's best to stick to this convention throughout this project for consistency.

MaxWasUnavailable replied:
> Also just noticed theres a typo 😄. It's supposed to be patch, not path.

Do you want me to fix this in this PR?

Kesomannen reviewed (commented):
Looks good, but there could be problems with merging the other PR, because I have edited some files that are also edited here. Maybe we'll just handle it when it's time to merge.

Also I should probably commit my changes there, I have a bunch of local changes.

MaxWasUnavailable committed:
Fix typo

As per https://github.com/MaxWasUnavailable/LethalAchievements/pull/6#discussion_r1574192833
